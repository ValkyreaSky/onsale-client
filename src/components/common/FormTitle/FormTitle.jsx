import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = styled.span`
  text-align: center;
  margin-bottom: 3rem;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 2.2rem;
`;

const Icon = styled.svg`
  width: 2.8rem;
  margin-right: 1rem;
`;

const FormTitle = ({ text, icon }) => (
  <Title>
    <Icon viewBox={icon.viewBox}>
      <use xlinkHref={icon} />
    </Icon>
    {text}
  </Title>
);

FormTitle.propTypes = {
  text: PropTypes.string.isRequired,
  icon: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default FormTitle;
export { Title, Icon };
