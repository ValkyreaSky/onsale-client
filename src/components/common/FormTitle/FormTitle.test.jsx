import React from 'react';
import { shallow } from 'enzyme';
import FormTitle, { Title, Icon } from './FormTitle';

describe('<FormTitle />', () => {
  const text = 'Hello';
  const icon = {
    viewBox: '0 0 489 323',
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<FormTitle text={text} icon={icon} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Title /> component with icon and text', () => {
    const wrapper = shallow(<FormTitle text={text} icon={icon} />);
    expect(wrapper.find(Title).length).toBe(1);
    expect(wrapper.find(Icon).length).toBe(1);
    expect(wrapper.find(Title).props().children[1]).toBe(text);
  });
});
