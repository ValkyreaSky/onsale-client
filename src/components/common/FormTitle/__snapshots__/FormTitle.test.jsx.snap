// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormTitle /> should match snapshot 1`] = `
<styled.span>
  <styled.svg
    viewBox="0 0 489 323"
  >
    <use
      xlinkHref={
        Object {
          "viewBox": "0 0 489 323",
        }
      }
    />
  </styled.svg>
  Hello
</styled.span>
`;
